types/react-relay/legacy.d.ts

```diff
@@ -94,36 +91,34 @@ export type RelayRefetchContainer<TComponent extends React.ElementType> = React.
     ContainerProps<React.ComponentPropsWithoutRef<TComponent>>
 >;
 
-export function createFragmentContainer<Props>(
-    Component: React.ComponentType<Props & { relay?: RelayProp | undefined }>,
-    fragmentSpec: Record<string, GraphQLTaggedNode>,
-): Container<Props>;
+type PropsWithoutRelay<C extends keyof JSX.IntrinsicElements | React.JSXElementConstructor<any>> =
+    JSX.LibraryManagedAttributes<C, Omit<React.ComponentProps<C>, 'relay'>>;
+
+export function createFragmentContainer<
+    C extends React.ComponentType<React.ComponentProps<C> & { relay?: RelayProp | undefined }>,
+>(Component: C, fragmentSpec: Record<string, GraphQLTaggedNode>): Container<PropsWithoutRelay<C>>;
 
 export { fetchQuery_DEPRECATED } from 'relay-runtime';
 
 export { graphql } from 'relay-runtime';
 export { readInlineData } from 'relay-runtime';
 export { requestSubscription } from 'relay-runtime';
 
-export function createPaginationContainer<Props>(
-    Component: React.ComponentType<
-        Props & {
-            relay: RelayPaginationProp;
-        }
-    >,
+export function createPaginationContainer<
```

Comment:
can we leave the formatting of this identical such that the only visible change in this file is the addition of `PropsWithoutRelay` passed to the generic of `Container`?

Reply:
I _can_ revert these changes, however they were only applied because I followed the contribution instructions to run prettier over any committed changes -- it looks like this may have been missed by previous contributors.

It seems to me that it would be less friction to apply these formatting changes now, or the same conversation will happen on the next commit.

Reply:
Or by formatting, do you mean only have a generic for `Props`? If so, I'm not sure I can achieve the proper handling of `defaultProps` with that as the only generic as I need a reference to the underlying component as a type rather than the extracted props.

Reply:
ah maybe yeah! True. I think im happy with this change, ill approve 1 moment.

Reply:
Thanks for the review @maraisr